File: applicationContext/WatchOS2LastEmoji/WatchOS2LastEmoji WatchKit Extension/InterfaceController.m
//
//  InterfaceController.m
//  WatchOS2LastEmoji WatchKit Extension
//
//  Created by Thai, Kristina on 6/21/15.
//  Copyright © 2015 Kristina Thai. All rights reserved.
//

#import "InterfaceController.h"
#import <WatchConnectivity/WatchConnectivity.h>

@interface InterfaceController() <WCSessionDelegate>

@property (strong, nonatomic) WCSession *session;

@end


@implementation InterfaceController

-(instancetype)init {
    self = [super init];
    
    if (self) {
        if ([WCSession isSupported]) {
            self.session = [WCSession defaultSession];
            self.session.delegate = self;
            [self.session activateSession];
        }
    }
    return self;
}

- (void)awakeWithContext:(id)context {
    [super awakeWithContext:context];

    // Configure interface objects here.
}

- (void)willActivate {
    // This method is called when watch view controller is about to be visible to user
    [super willActivate];
}

- (void)didDeactivate {
    // This method is called when watch view controller is no longer visible
    [super didDeactivate];
}
- (IBAction)catPressed {
    [self sendEmoji:@"cat"];
}
- (IBAction)dogPressed {
    [self sendEmoji:@"dog"];
}
- (IBAction)pandaPressed {
    [self sendEmoji:@"panda"];
}
- (IBAction)bunnyPressed {
    [self sendEmoji:@"bunny"];
}
- (IBAction)cowPressed {
    [self sendEmoji:@"cow"];
}
- (IBAction)hamsterPressed {
    [self sendEmoji:@"hamster"];
}

-(void)sendEmoji:(NSString *)emoji {
    NSDictionary *applicationDict = @{@"emoji":emoji};
    [self.session updateApplicationContext:applicationDict error:nil];
}

@end

File: WatchOS2LastEmojiSwift/WatchOS2LastEmojiSwift WatchKit Extension/InterfaceController.swift
//
//  InterfaceController.swift
//  WatchOS2LastEmojiSwift WatchKit Extension
//
//  Created by Thai, Kristina on 6/21/15.
//  Copyright © 2015 Kristina Thai. All rights reserved.
//

import WatchKit
import Foundation
import WatchConnectivity


class InterfaceController: WKInterfaceController, WCSessionDelegate {
    
    var session : WCSession!

    override func awakeWithContext(context: AnyObject?) {
        super.awakeWithContext(context)
    }

    override func willActivate() {
        super.willActivate()
        
        if (WCSession.isSupported()) {
            session = WCSession.defaultSession()
            session.delegate = self
            session.activateSession()
        }
    }

    override func didDeactivate() {
        super.didDeactivate()
    }
    
    @IBAction func catPressed() {
        sendEmoji("cat")
    }
    
    @IBAction func dogPressed() {
        sendEmoji("dog")
    }

    @IBAction func pandaPressed() {
        sendEmoji("panda")
    }
    
    @IBAction func bunnyPressed() {
        sendEmoji("bunny")
    }
    
    @IBAction func cowPressed() {
        sendEmoji("cow")
    }
    
    @IBAction func hamsterPressed() {
        sendEmoji("hamster")
    }
    
    func sendEmoji(emoji: String){
        let applicationDict = ["emoji":emoji]
        do {
            try session.updateApplicationContext(applicationDict)
        } catch {
            print("error")
        }
    }
}
